### ISSUE TITLE
[Accessibility] Narrator doesn't announce check/unchecked state for DateTimePicker with ShowCheckBox property enabled

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#10363]
### .NET version

9.0.100-alpha.1.23564.26

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, reproduce in previous .Net versions

### Issue description

When dataTimePicker's ShowCheckBox property enabled, Narrator doesn't announce check/unchecked state for DateTimePicker if check/uncheck its checkbox
![image](https://github.com/dotnet/winforms/assets/26474449/e5b54a87-5fc9-4fbf-a3f5-48486039464a)
![DateTimePicker](https://github.com/dotnet/winforms/assets/26474449/f3579ea3-5a5f-4280-9746-448fba3bb0eb)

### Steps to reproduce

Test application: 
[DateTimePickerSample.zip](https://github.com/dotnet/winforms/files/13446509/DateTimePickerSample.zip)

1. Build and run test application
2. Open Narrator, tab to DateTimePicker, press space to check/uncheck state, then observe

**Comment by merriemcgaw** [comment]
This is external to Win32
